site/zenodo_rdm/assets/semantic-ui/js/zenodo_rdm/src/support/components/CategoryDropdown.js

```diff
@@ -0,0 +1,69 @@
+import React, { Component } from "react";
+import { PropTypes } from "prop-types";
+import { SelectField } from "react-invenio-forms";
+import _find from 'lodash';
+
+class CategoryDropdown extends Component {
+
+    constructor(props) {
+        super(props);
+        let activeCategory = null;
+        if (Array.isArray(props.categories)) {
+            activeCategory = this.props.categories.find((el) => (el.key === props.defaultCategory))
+        }
+
+        this.state = {
+            activeCategory: activeCategory,
+        }
+    }
+
+    serializeCategory(category) {
+        if (!category) {
+            return null;
+        }
+        return {
+            text: category.title,
+            value: category.key
+        }
+    }
+
+    onChange = ({ data, formikProps }) => {
+        const activeCategory = this.props.categories.find((el) => (el.key === data.value))
+        this.setState({
+            activeCategory: activeCategory
+        });
+        formikProps.form.setFieldValue('category', data.value);
+    }
```

Comment:
```suggestion
        const { categories } = this.props;
        const activeCategory = categories.find((el) => (el.key === data.value))
        
        this.setState({
            activeCategory: activeCategory
        });
        formikProps.form.setFieldValue('category', data.value);
    }
```

Reply:
minor: this can be simplified to 
```suggestion
        ...
        this.setState({activeCategory});
        ...    
```